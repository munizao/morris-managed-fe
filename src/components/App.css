main {
  background-color: #2c2b26;
  background-attachment: fixed;
  background-size: cover;
  min-height: calc(100vh - 4rem);
}