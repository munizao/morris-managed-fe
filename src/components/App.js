import React, { useState, useEffect } from 'react';
import CompetencyContainer from './CompetencyView/CompetencyContainer';
import Header from './Header';
import { login, signedIn, logout } from '../services/mm';
import './App.css';


export default () => {
  const [user, setUser] = useState({
    isSignedIn: false,
    dancerId: '',
    name: '',
  });

  const [authInputs, setAuthInputs] = useState({ email: '', password: '' });

  useEffect(() => {
    signedIn()
      .then((res) => {
        console.log(res);
        if(res.status) {
          setUser({ isSignedIn: false });
        }
        else {
          setUser({ isSignedIn: true, dancerId: res.dancer._id, name: res.dancer.name });
        }
      });
  }, []);

  const handleAuthSubmit = (event) => {
    event.preventDefault();
    login(authInputs.email, authInputs.password)
      .then((res) => {
        if(res.status) {
          console.log(res); //TODO: handle case of invalid login
        }
        else {
          setUser({ isSignedIn: true, name: res.dancer.name, dancerId: res.dancer._id });
        }
      });
  };

  const handleInputChange = ({ target }) => {
    setAuthInputs({ [target.name]: target.value });
  };

  const handleLogoutClick = () => {
    logout()
      .then(() => {
        setUser({ isSignedIn: false });
      });
  };

  return (
    <>
      <Header isSignedIn={user.isSignedIn} 
        dancerName={user.name} 
        authSubmit={handleAuthSubmit} 
        inputChange={handleInputChange} 
        logoutClick={handleLogoutClick} />
      <main>
        <CompetencyContainer />
      </main>
    </>
  );
};

