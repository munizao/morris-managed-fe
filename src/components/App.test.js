import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

jest.mock('../services/mm.js', () => ({
  signedIn: jest.fn(() => Promise.resolve()),
  allDances: jest.fn(() => Promise.resolve())
}));

describe('App component', () => {
  it('renders App', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });
});
