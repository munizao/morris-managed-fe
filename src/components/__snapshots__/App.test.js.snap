// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders App 1`] = `
<Fragment>
  <Header
    authSubmit={[Function]}
    dancerName=""
    inputChange={[Function]}
    isSignedIn={false}
    logoutClick={[Function]}
  />
  <main>
    <CompetencyContainer />
  </main>
</Fragment>
`;
